torchtext/csrc/vocab.cpp

```diff
@@ -20,24 +20,24 @@ Vocab::Vocab(const StringList &tokens, const std::string &unk_token)
       throw std::runtime_error("Duplicate token found in tokens list: " +
                                tokens[i]);
     }
-    stoi_.insert(std::move(tokens[i]), i);
+    stoi_[std::move(tokens[i])] = i;
   }
-  unk_index_ = stoi_.find(unk_token)->value();
+  unk_index_ = stoi_.find(unk_token)->second;
 }
 
 int64_t Vocab::__len__() const { return stoi_.size(); }
 
 int64_t Vocab::__getitem__(const std::string &token) const {
   const auto &item = stoi_.find(token);
   if (item != stoi_.end()) {
-    return item->value();
+    return item->second;
   }
   return unk_index_;
 }
 
 void Vocab::append_token(const std::string &token) {
   if (stoi_.find(token) == stoi_.end()) {
-    stoi_.insert(std::move(token), stoi_.size());
+    stoi_[std::move(token)] = stoi_.size();
   }
```

Comment:
I guess we do need to update `itos` func here.

Reply:
Good catch!